### ISSUE TITLE
Feature Request: Prioritize Deletion of Older Nodes During Scale Down in v0.34.0+ version

### ISSUE THREAD

**Issue by Akash23678** [aws/karpenter-provider-aws#6590]
We need a feature in Karpenter to prioritize the termination of older nodes over newer ones during scale down. This is crucial for our operations, particularly when updating EKS clusters to new versions, as it ensures minimal disruption and better resource optimization.

When updating our EKS cluster from version 1.27 to 1.29, we need to replace older nodes with new nodes running the updated version. Prioritizing the deletion of older nodes would streamline this process.

We can delete older (1.27) version node explicitly by gracefull termination, but it's difficult when you are dealing with 50-100+ nodes.

**Comment by engedaam** [comment]
Can you help me understand your use case more? Why does expireAfter not work for your use case?

**Comment by Akash23678** [comment]
expireAfter is like replacement of new nodes after given time interval. So my use case is, I want to replace all older k8s version node with new k8s version while cluster upgrade.

If we have some feature like while scaling down nodes, always delete older age node, which will help us to update older version node with newer version node.
Here version means k8s version.

**Comment by jukie** [comment]
Wouldn't enabling drift handle the case of upgrading nodes and only having the older nodes disrupted?

**Comment by Akash23678** [comment]
I don't think drifted will works here, it's for different usecases.

**Comment by engedaam** [comment]
> When updating our EKS cluster from version 1.27 to 1.29, we need to replace older nodes with new nodes running the updated version.

@Akash23678 For upgrading nodes after a cluster upgrade, Are you using automatic AMI discovery? If so, the nodes should have been drifted

**Comment by jukie** [comment]
@Akash23678 Drift definitely works for detecting AMI drift after an upgrade. I do still like the idea of adding a feature to influence the decision of or somehow prioritize which nodes are deleted first (besides things like pod-deletion-cost annotations on pods).

**Comment by Akash23678** [comment]
But somehow we disable drift due some internal policies.
But drift things applies on newly node launched right?
How come drift will apply on older/existing node to delete, I didn't understand.

**Comment by Akash23678** [comment]
> > When updating our EKS cluster from version 1.27 to 1.29, we need to replace older nodes with new nodes running the updated version.
> 
> @Akash23678 For upgrading nodes after a cluster upgrade, Are you using automatic AMI discovery? If so, the nodes should have been drifted

@engedaam we are not using ami discovery

**Comment by FernandoMiguel** [comment]
> @engedaam we are not using ami discovery

so when you upgrade clusters, right after that you update your nodepool AMI ?

**Comment by Akash23678** [comment]
> > @engedaam we are not using ami discovery
> 
> so when you upgrade clusters, right after that you update your nodepool AMI ?

Yes, that's how, we have been doing update AMI to new k8s version in nodepool/awsnodetmplate.

**Comment by FernandoMiguel** [comment]
And that means that new nodes have the new AMI, but since you have drift disabled, the old ones don't get replaced.
Right?
Have you considered enabling drift as part of the upgrade process and then turning it off again?
Should address the automation of replacing old nodes, instead of manually removing them...
